osu.Game/Rulesets/Edit/Checks/CheckConcurrentObjects.cs

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Collections.Generic;
+using osu.Game.Beatmaps;
+using osu.Game.Rulesets.Edit.Checks.Components;
+using osu.Game.Rulesets.Objects;
+using osu.Game.Rulesets.Objects.Types;
+
+namespace osu.Game.Rulesets.Edit.Checks
+{
+    public class CheckConcurrentObjects : ICheck
+    {
+        // We guarantee that the objects are either treated as concurrent or unsnapped when near the same beat divisor.
+        private const double ms_leniency = CheckUnsnappedObjects.UNSNAP_MS_THRESHOLD;
+
+        public CheckMetadata Metadata { get; } = new CheckMetadata(CheckCategory.Compose, "Concurrent hitobjects");
+
+        public IEnumerable<IssueTemplate> PossibleTemplates => new IssueTemplate[]
+        {
+            new IssueTemplateConcurrentSame(this),
+            new IssueTemplateConcurrentDifferent(this)
+        };
+
+        public IEnumerable<Issue> Run(IBeatmap playableBeatmap, IWorkingBeatmap workingBeatmap)
+        {
+            for (int i = 0; i < playableBeatmap.HitObjects.Count - 1; ++i)
+            {
+                var hitobject = playableBeatmap.HitObjects[i];
+
+                for (int j = i + 1; j < playableBeatmap.HitObjects.Count; ++j)
+                {
+                    var nextHitobject = playableBeatmap.HitObjects[j];
+
+                    // Accounts for rulesets with hitobjects separated by columns, such as Mania.
+                    // In these cases we only care about concurrent objects within the same column.
+                    if ((hitobject as IHasColumn)?.Column != (nextHitobject as IHasColumn)?.Column)
+                        continue;
```

Comment:
That was my previous implementation for this. Reason I changed was to make it similarly generic to the unsnapped check, which uses `IHasRepeats` instead of having a `CheckUnsnappedReverses`, for example.

No matter what ruleset we're talking about, if it uses columns then this check should apply per-column is the idea, and the current code expresses that much better than splitting out the check across every single ruleset imo.

Any specific reason to have custom rulesets need to implement their own version of `IHasColumn`?

Reply:
Hmm, I suppose it's fine... Rulesets can do whatever they want. They aren't required to use `IHasColumn`, and neither required to use any of our other interfaces. But generally we expose such interfaces when they have a purpose, and I guess this gives it a purpose?